src/component-library/Input/Input.style.tsx

```diff
@@ -35,25 +42,35 @@ const BaseInput = styled.input`
   }
 `;
 
-type WrapperProps = {
-  $hasStartAdornment: boolean;
-  $hasEndAdornment: boolean;
+type BaseInputWrapperProps = {
+  $hasStartAdornment?: boolean;
+  $hasEndAdornment?: boolean;
+  $hasError?: boolean;
+  $isDisabled?: boolean;
 };
 
-const Wrapper = styled.div<WrapperProps>`
-  box-shadow: ${theme.boxShadow.default};
+const BaseInputWrapper = styled.div<BaseInputWrapperProps>`
+  background-color: ${theme.input.background};
   border-radius: ${theme.rounded.md};
-  border: ${theme.border.default};
   color: ${theme.colors.textPrimary};
   box-sizing: border-box;
   display: flex;
   align-items: center;
   padding-left: ${(props) => props.$hasStartAdornment && theme.spacing.spacing2};
   padding-right: ${(props) => props.$hasEndAdornment && theme.spacing.spacing2};
+  border: ${(props) => {
+    if (props.$isDisabled) return theme.input.disabled.border;
```

Comment:
Again, I think I'd find a ternary cleaner here. Even checking the two conditions, I think a chained ternary is ok.